### ISSUE TITLE
Integrated Dictionary Lookup Feature in AndBible

### ISSUE THREAD

**Issue by bvahedy** [AndBible/and-bible#1969]
**Title:** Request for Integrated Dictionary Lookup Feature in AndBible

**Summary:**
A feature request was made to allow users to look up words (e.g., names like "Aaron") directly in Bible dictionaries within AndBible. Discussions included possible implementations and dictionary options, but no confirmation of adoption was given.

**Key Points:**

* User suggests the ability to highlight a word and look it up in Bible dictionaries like *Smith's* directly from the app.
* agrogers and bvahedy note similar past proposals but couldn't locate an existing issue.
* agrogers suggests a form listing all available dictionaries containing the word; user selects the desired one, opening it in a maximized or existing window.
* tuomas2 raises the issue of dictionary availability as Sword modules, particularly noting the lack of a general English dictionary module.
* agrogers clarifies the preference for Bible-specific dictionaries (*Smith's, Vine's, ISBE*).
* Proposed implementation: long-click word → select “View dictionary definitions” from context menu → open multi-doc window with definitions from configured dictionaries.
* timbze mentions *Webster 1828* dictionary as another option.
* The feature is seen as part of a potential larger "word study" feature with a dedicated UI, similar to how commentaries are presented in AndBible (#1970).

****

<details><summary>Original message</summary>
<p>
User requested the following:

From ticket:
https://support.andbible.org/scp/tickets.php?id=1100

Do you think that AndBible might have this as a feature in a future version? I can see a good application for being able to look up a word in a module for example if you would like to use a specialized dictionary like a Bible dictionary. You might highlight the name "Aaron" and then easily look that name up in Smith's Bible dictionary or something like that.
</p>
</details>

**Comment by agrogers** [comment]
This was my proposal to address this request. I am sure it already in another issue somewhere but i couldnt find it.
https://github.com/AndBible/and-bible/issues/1970

**Comment by bvahedy** [comment]
Yes, I tried also to look for it before creating a ticket but couldn't find
it either.

**Comment by tuomas2** [comment]
Is there a suitable dictionary modle available?

**Comment by agrogers** [comment]
All or any dictionary would be fine for me. Obviously some words will not
be included in some or all dictionaries.

My idea has been that the form will list all the dictionaries that the word
is available in. The user clicks the dictionary they want and it opens it
in a new maximised window (pushing the back button would close the window)
or in an existing window of that document type.

___________________________
*www.TenOutOfTen.org* <https://www.TenOutOfTen.org>
***@***.***
(+61) 0477 174 193 (Australia)
(+95) 09 250018669 (Myanmar)

**Comment by tuomas2** [comment]
Yes but is there a proper english dictionary as Sword module, for this feature to be useful?

**Comment by agrogers** [comment]
Oh, I don't want a normal dictionary usually. I want Smiths or Vines or
ISBE or any of the others.

**Comment by tuomas2** [comment]
My proposal for implementation:

long-click word -> from context popup menu "View dictionary definitions" (shortly: "Dictionaries")
-> opens a new multi-doc window that displays dict definitions from chosen dictionaries (can be configured in app settings).

**Comment by timbze** [comment]
I like Webster 1828 which is also available

**Comment by agrogers** [comment]
> My proposal for implementation:
> 
> long-click word -> from context popup menu "View dictionary definitions" (shortly: "Dictionaries") -> opens a new multi-doc window that displays dict definitions from chosen dictionaries (can be configured in app settings).

I think the multi-doc window is great but getting to that window via a context menu would be an interim step. The end goal should be something more like #1970 perhaps with the dictionaries listed like the commentaries are listed. Or put differently, a word study is something that could (should?) be an important feature and as such probably needs a more comprehensive UI.

**Comment by tuomas2** [comment]
Estimate 3h

**Comment by meichthys** [comment]
@tuomas2 Can you clarify if this estimate is for your proposal here: https://github.com/AndBible/and-bible/issues/1969#issuecomment-989821489

Or for the proposal here: https://github.com/AndBible/and-bible/issues/1969#issuecomment-990662989

**Comment by tuomas2** [comment]
Estimate is for my proposal.
